src/assets/scripts/client/aircraft/FlightManagementSystem/Fms.js

```diff
@@ -699,11 +703,6 @@ export default class Fms {
      * @param runwayModel {RunwayModel}
      */
     replaceDepartureProcedure(routeString, runwayModel) {
-        // this is the same procedure that is already set, no need to continue
-        if (this.hasLegWithRouteString(routeString)) {
```

Comment:
>But, I need to ask; what if you have the same routeString and the same runwayModel?

Actually, we can address this without messing any of that up by having `.setDepartureRunway()` return early if the runway being "set" is the same as the one already in there. Then, `.replaceDepartureProcedure()` won't be called. And we can continue with the conservative approach of having `.replaceDepartureProcedure()` freshly regenerating the waypoints. Sounds to me like the best solution.